lib/oanda_api/client/resource_descriptor.rb

```diff
@@ -43,10 +49,25 @@ def is_collection?
       # @param [String] resource_name
       # @return [void]
       def resource_klass=(resource_name)
-        klass_symbol = resource_name.capitalize.to_sym
+        klass_symbol = OandaAPI::Utils.classify(resource_name).to_sym
         fail ArgumentError, "Invalid resource" unless OandaAPI::Resource.constants.include?(klass_symbol)
         @resource_klass = OandaAPI::Resource.const_get klass_symbol
       end
+
+      # Will set instance attributes based on resource_name, method and resource_id.
+      #
+      # @param [String] resource_name name of the resource (from the path).
+      # @param [Symbol] method an http verb (see {OandaAPI::Client.map_method_to_http_verb}).
+      # @param [Symbol] resource_id optional id of the resource.
+      # @return [void]
+      def initialize_from_resource_name(resource_name, method, resource_id: nil)
+        mapped_resource = RESOURCES_MAPPER[resource_name.to_sym]
```

Comment:
I'm thinking this method could be simplified slightly like this:
```
        mapped_resource =
          RESOURCES_MAPPER.fetch(resource_name.to_sym,
                                 { resource_name: Utils.singularize(resource_name),
                                   is_collection: method == :get && resource_id.empty?
                                 })
        self.resource_klass = mapped_resource.fetch :resource_name
        @is_collection = mapped_resource.fetch :is_collection
        @collection_name = Utils.pluralize(resource_name).to_sym if is_collection?
```
I like using `#fetch` for hash keys that should exist. For example, both of the keys `:resource_name` and `:is_collection` should exist in every RESOURCES_MAPPER entry.